### ISSUE TITLE
About OSEDiff and Hypir

### ISSUE THREAD

**Issue by lhao0301** [XPixelGroup/HYPIR#23]
It seems that `hypir` performs better than `osediff`. I wonder what's the core difference that cause the different performance on test images.
Take sd2.1 based model for example (osddiff also is based on sd2.1),  gan-loss(vs vsd-loss?) or training data(beyond lsdir?) or some others important factors. 
Could u give some suggestions.

**Comment by 0x3f3f3f3fun** [comment]
I think the core difference lies in the GAN-loss and VSD-loss.

The core idea of VSD-loss ([OSEDiff](https://arxiv.org/abs/2406.08177), [DMD](https://arxiv.org/abs/2311.18828), [DI](https://arxiv.org/abs/2305.18455)) is that to make two distributions closer, you can make the diffusion models (i.e. score functions) trained on those two distributions closer. OSEDiff uses a "regularizer" to learn the restoration network's output distribution and then makes this regularizer close to a pretrained diffusion model (the teacher model). This forces the network's output distribution to align with the pretrained data distribution. Since its goal is the pretrained distribution, its performance is ultimately bounded by the teacher model.

With GAN-loss, the discriminator is trained dynamically. If we use very high-quality training data, the discriminator will guide the restoration network to produce equally high-quality output. Therefore, the performance upper-bound is determined by the data used for training.

Based on my observations, when both are trained with LSDIR, OSEDiff's style is smoother, just like a diffusion model, while GAN-based methods (S3Diff, HYPIR-SD2) are sharper.

**Comment by lhao0301** [comment]
Thanks for your advice. Will the pretrained `D` net (actually, the decoder ) be released?

**Comment by 0x3f3f3f3fun** [comment]
Released [now](https://huggingface.co/lxq007/HYPIR/tree/main). How to use:

```python
from safetensors.torch import load_file
D.decoder.load_state_dict(load_file("path/to/HYPIR_sd2_D.safetensors"))
```
